ampersand-select-view.js

```diff
@@ -359,15 +359,26 @@ module.exports = View.extend({
             if (!model) throw new Error('model or model idAttribute not found in options collection');
             return this.yieldModel ? model : model[this.idAttribute];
         } else if (Array.isArray(this.options)) {
-            // find value value in options array
-            // find option, formatted [['val', 'text'], ...]
-            if (this.options.length && Array.isArray(this.options[0])) {
-                for (var i = this.options.length - 1; i >= 0; i--) {
-                    if (this.options[i][0] == value) return this.options[i];
+            var i = this.options.length - 1;
+            if (i >= 0) {
+                // find value in collection as [['val','text'], ...]
+                if (Array.isArray(this.options[0])) {
+                    for (i; i >= 0; i--) {
+                        if (this.options[i][0] == value) {
+                            return this.options[i][0];
+                        }
+                    }
+                // find value in collection as ['val', ...]
+                } else {
+                    for (i; i >= 0; i--) {
+                        if (this.options[i] == value) {
+                            return this.options[i];
+                        }
+                    }
                 }
+            } else {
+                throw new Error('no options provided');
```

Comment:
I like and support this, we tend to want to throw exceptions when devs get themselves into places that would give unexpected results.  This will mean a major release though because it would not have thrown in this case before.